plugins/com.aptana.editor.common/src/com/aptana/editor/common/CommonUtil.java

```diff
@@ -31,4 +48,46 @@ public static IToken getToken(String tokenName)
 		return new Token(tokenName);
 	}
 
+	/**
+	 * Registering double click listener to the project explorer viewer to handle
+	 * "Open no extension files with JavaScript Source Editor" preference option. Which is added under the "Editors"
+	 * section. Based on the preference option while opening a file will overwrite the default editor id of a file.
+	 */
+	public static void handleOpenWithEditorPref()
+	{
+		if (!isDoubleClickRegistered)
+		{
+			IViewPart explorerPart = UIUtils.findView(IPageLayout.ID_PROJECT_EXPLORER);
+			if (explorerPart != null)
+			{
+				CommonNavigator projectExplorer = (CommonNavigator) explorerPart;
+				CommonViewer viewer = projectExplorer.getCommonViewer();
+				viewer.addDoubleClickListener(new IDoubleClickListener()
+				{
+					public void doubleClick(DoubleClickEvent event)
+					{
+						IResource selectedResource = UIUtils.getSelectedResource();
+						if (selectedResource != null && selectedResource instanceof IFile)
+						{
```

Comment:
Just being more nitty-gritty. The code will be more readable if we can avoid nested 'if' conditions. We have a common practice to check for a condition and return early, something on these lines:

```
if (selectedResource == null || !(selectedResource instanceof IFile))
{
  return;
}
```